mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java

```diff
@@ -50,9 +51,30 @@ void configureTestRunner(Model model) {
                     .collect(Collectors.toList()).toString());
 
             dependencyResolver.addRequiredDependencies(model);
+            effectiveTestRunnerPluginConfigurations.forEach(this::configurePlugin);
+        }
+    }
+
+    private void configurePlugin(Plugin testRunnerPlugin) {
+            dependencyResolver.addAsPluginDependency(testRunnerPlugin);
+
+            if (!this.configuration.isSelectingMode()) {
+                final List<PluginExecution> executions = testRunnerPlugin.getExecutions();
+                executions.forEach(this::addSkipAfterFailureCount);
+            }
+    }
 
-            effectiveTestRunnerPluginConfigurations
-                .forEach(dependencyResolver::addAsPluginDependency);
+    private void addSkipAfterFailureCount(PluginExecution pluginExecution) {
+        final Xpp3Dom pluginConfiguration = getOrCreatePluginConfiguration(pluginExecution);
+        final String skipAfterFailureCountProperty = "skipAfterFailureCount";
+        final Xpp3Dom count = pluginConfiguration.getChild(skipAfterFailureCountProperty);
+        if (count == null) {
+            final Xpp3Dom xpp3Dom = new Xpp3Dom(skipAfterFailureCountProperty);
+            final String skipAfterFailureCount = System.getProperty("surefire.skipAfterFailureCount");
+            final String skipAfterFailureCountValue = skipAfterFailureCount != null ? skipAfterFailureCount : "1";
```

Comment:
This can be oneliner `System.getProperty("surefire.skipAfterFailureCount", "1")`